### PR TITLE
Node: Add adminserver command to generate template for accountant modify balance

### LINKED ISSUE #3902: Node / Admin: Missing template command for AccountantModifyBalance
Although the guardian supports consuming this admin command, there is no command for generating the prototext for it. That should be added.

### PR DESCRIPTION
**Pull request by bruce-riley**
Somehow this governance action got added without a command to generate the prototext. This PR adds that command.

Fixes #3902.

### CODE REVIEW

File: node/cmd/guardiand/admintemplate.go [hunk 1]
```diff
@@ -495,6 +527,103 @@ func runRecoverChainIdTemplate(cmd *cobra.Command, args []string) {
...
+	}
+	_, overflow := uint256.FromBig(amount_big)
+	if overflow {
+		log.Fatal("amount overflowed uint256")
+	}
+	if *accountantModifyBalanceReason == "" {
+		log.Fatal("--reason must be specified.")
+	}
+	if len(*accountantModifyBalanceReason) > 32 {
+		log.Fatal("reason is too long, can be at most 32 bytes")
```
Comment (SEJeff) [inline node/cmd/guardiand/admintemplate.go:595]: Can any or all of these numbers be constants? Should they be?

File: node/cmd/guardiand/admintemplate.go [hunk 2]
```diff
@@ -177,6 +188,21 @@ func init() {
...
+	accountantModifyBalanceFlagSet := pflag.NewFlagSet("accountant-modify-balance", pflag.ExitOnError)
+	accountantModifyBalanceModule = accountantModifyBalanceFlagSet.String("module", "GlobalAccountant", "Module identifier of the accountant")
+	accountantModifyBalanceTargetChainId = accountantModifyBalanceFlagSet.String("target-chain-id", "", "ID of the chain to receive this modify")
+	accountantModifyBalanceSequence = accountantModifyBalanceFlagSet.String("sequence", "", "The sequence number of this modification.  Each modification must be uniquely identifiable just by its sequence number")
+	accountantModifyBalanceChainId = accountantModifyBalanceFlagSet.String("chain-id", "", "Chain ID of the account to be modified")
+	accountantModifyBalanceTokenChainId = accountantModifyBalanceFlagSet.String("token-chain-id", "", "Chain ID of the native chain for the token")
+	accountantModifyBalanceTokenAddress = accountantModifyBalanceFlagSet.String("token-address", "", "Address of the token on its native chain, hex string encoded")
+	accountantModifyBalanceKind = accountantModifyBalanceFlagSet.String("action", "", "Kind of modification to be made (1 = add, 2 = sub)")
```
Comment (panoel) [inline node/cmd/guardiand/admintemplate.go:199]: Maybe accountantModifyBalanceAction is a more appropriate name?  IDK?

File: node/cmd/guardiand/admintemplate.go (hunk 1 above)
Comment (bruce-riley) [reply node/cmd/guardiand/admintemplate.go:595]: I added a length for the reason. I think that's the only one here for which that makes sense. Could maybe do another PR some day to add consts for all the other commands in `sdk/vaa/payload.go`, but that's beyond the scope of this PR.

**Review by panoel — DISMISSED** [review]

File: node/cmd/guardiand/admintemplate.go [hunk 3]
```diff
@@ -177,6 +188,21 @@ func init() {
 	AdminClientRecoverChainIdCmd.Flags().AddFlagSet(moduleFlagSet)
 	TemplateCmd.AddCommand(AdminClientRecoverChainIdCmd)
 
+	// flags for the accountant-modify-balance command
+	accountantModifyBalanceFlagSet := pflag.NewFlagSet("accountant-modify-balance", pflag.ExitOnError)
+	accountantModifyBalanceModule = accountantModifyBalanceFlagSet.String("module", "GlobalAccountant", "Module identifier of the accountant")
+	accountantModifyBalanceTargetChainId = accountantModifyBalanceFlagSet.String("target-chain-id", "", "ID of the chain to receive this modify")
```
Comment (SEJeff) [inline node/cmd/guardiand/admintemplate.go:194]:
```suggestion
	accountantModifyBalanceTargetChainId = accountantModifyBalanceFlagSet.String("target-chain-id", "", "ID of the chain to receive this modification")
```

File: node/cmd/guardiand/admintemplate.go [hunk 4]
```diff
@@ -495,6 +527,103 @@ func runRecoverChainIdTemplate(cmd *cobra.Command, args []string) {
...
+						Reason:        *accountantModifyBalanceReason,
+					},
+				},
+			},
+		},
+	}
+
+	b, err := prototext.MarshalOptions{Multiline: true}.Marshal(m)
+	if err != nil {
+		panic(err)
```
Comment (SEJeff) [inline node/cmd/guardiand/admintemplate.go:622]: Can this be a log.Fatal() with a bit more context on what broke please?

File: node/cmd/guardiand/admintemplate.go [hunk 5]
```diff
@@ -1098,9 +1227,9 @@ func parseChainID(name string) (vaa.ChainID, error) {
 	}
 
 	// parse as uint32
-	i, err := strconv.ParseUint(name, 10, 32)
+	i, err := strconv.ParseUint(name, 10, 16)
```
Comment (SEJeff) [inline node/cmd/guardiand/admintemplate.go:1230]: Should that comment be updated to `// parse as uint16` now that you changed it? This does make so much more sense too as chains are uint16s. Good change.

File: node/pkg/adminrpc/adminserver.go [hunk 6]
```diff
@@ -251,8 +251,8 @@ func accountantModifyBalance(req *nodev1.AccountantModifyBalance, timestamp time
 		return nil, errors.New("invalid new token address (expected 32 bytes)")
 	}
 
-	if len(req.Reason) > 32 {
-		return nil, errors.New("the reason should not be larger than 32 bytes")
+	if len(req.Reason) > vaa.AccountantModifyBalanceReasonLength {
+		return nil, fmt.Errorf("the reason should not be larger than %d bytes", vaa.AccountantModifyBalanceReasonLength)
 	}
```
Comment (SEJeff) [inline node/pkg/adminrpc/adminserver.go:256]: Great use of a constant. We find so many (security or non-security) bugs related to not-using constants. You'd actually be surprised!

File: node/pkg/adminrpc/prototext_test.go [hunk 7]
```diff
@@ -536,7 +536,7 @@ var adminCommandTest = []adminCommandTestEntry{
 						chain_id: 1
 						token_chain: 2
 						token_address: "000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2"
-						kind: 1
+						kind:  MODIFICATION_KIND_ADD
 						amount: "115792089237316195423570985008687907853269984665640564039457584007913129639936"
 						reason: "fix bad value"
 					}
```
Comment (SEJeff) [inline node/pkg/adminrpc/prototext_test.go:542]: Would it be too much to ask for a `MODIFICATION_KIND_SUB` test?

**Review by SEJeff — DISMISSED** [review]
Seems like a sensible feature and it is gated by governance. Nice

**Review by SEJeff — APPROVED** [review]

**Review by panoel — APPROVED** [review]

**Review by evan-gray — APPROVED** [review]

**Pull request merged**